PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx

```diff
@@ -129,131 +143,201 @@ struct HfCandidateCreatorSigmac0plusplus {
+  void createSigmaC(aod::Collisions::iterator const& collision,
+                    TRK const& trackSoftPi,
+                    aod::TracksWDcaExtra const& tracks,
+                    CandidatesLc const& candidates,
+                    aod::BCsWithTimestamps const& bcWithTimeStamps)
+  {
+
+    auto thisCollId = collision.globalIndex();
+
+    histos.fill(HIST("hCounter"), 2);
+    /// keep only soft-pion candidate tracks
+    /// if not selected, skip it and go to the next one
+    if (!softPiCuts.IsSelected(trackSoftPi)) {
+      return;
+    }
+    /// dcaXY, dcaZ selections
+    /// To be done separately from the others, because for reassigned tracks the dca must be recalculated
+    /// TODO: to be properly adapted in case of PV refit usage
+    if (trackSoftPi.collisionId() == thisCollId) {
+      /// this is a track originally assigned to the current collision
+      /// therefore, the dcaXY, dcaZ are those already calculated in the track-propagation workflow
+      if (std::abs(trackSoftPi.dcaXY()) > softPiDcaXYMax || std::abs(trackSoftPi.dcaZ()) > softPiDcaZMax) {
+        return;
+      }
+    } else {
+      /// this is a reassigned track
+      /// therefore we need to calculate the dcaXY, dcaZ with respect to this new primary vertex
+      auto bc = collision.bc_as<o2::aod::BCsWithTimestamps>();
+      initCCDB(bc, runNumber, ccdb, isRun2Ccdb ? ccdbPathGrp : ccdbPathGrpMag, lut, isRun2Ccdb);
+      auto trackParSoftPi = getTrackPar(trackSoftPi);
+      o2::gpu::gpustd::array<float, 2> dcaInfo{-999., -999.};
+      o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackParSoftPi, 2.f, noMatCorr, &dcaInfo);
+      if (std::abs(dcaInfo[0]) > softPiDcaXYMax || std::abs(dcaInfo[1]) > softPiDcaZMax) {
+        return;
+      }
+    }
+    histos.fill(HIST("hCounter"), 3);
+
+    /// loop over Λc+ → pK-π+ (and charge conj.) candidates
+    if constexpr (withTimeAssoc) {
+      /// need to group candidates manually
+      auto candidatesThisColl = candidates.sliceBy(hf3ProngPerCollision, thisCollId);
+      makeSoftPiLcPair(trackSoftPi, candidatesThisColl, tracks);
+    } else {
+      /// tracks and candidates already grouped by collision at the level of process function
+      makeSoftPiLcPair(trackSoftPi, candidates, tracks);
+    }
+
+  } /// end createSigmaC
+
+  /// @brief process function for Σc0,++ → Λc+(→pK-π+) π- candidate reconstruction considering also time-reassigned tracks for soft pions
+  /// @param collisions are o2::aod::Collisions
+  /// @param trackIndices are the indices of tracks reassigned to collisions, as obtained from the track-to-collision-associator
+  /// @param tracks are the tracks (with dcaXY, dcaZ information) → soft-pion candidate tracks
+  /// @param candidates are 3-prong candidates satisfying the analysis selections for Λc+ → pK-π+ (and charge conj.)
+  void processDataTimeAssoc(aod::Collisions const& collisions,
```

Comment:
I would change the name of this process function, since with the `trackToCollisionAssociation` one can also have the standard association and not the time based one. For instance you could use `processDataTrackToCollAssoc`